processing/src/test/java/org/apache/druid/segment/data/IncrementalIndexTest.java

```diff
@@ -725,4 +728,171 @@ public void testDynamicSchemaRollup() throws IndexSizeExceededException
 
     Assert.assertEquals(2, index.size());
   }
+
```

Comment:
do we need sad path tests here? like is it possible to end up with a mix of aggregators that are not the same type if the ingestion schema changed between segment creation? what if non-rolled up has columns with the same name as aggregators in rolled up? should these be an error? is there some external safeguards that prevent these tasks from ever getting started due to incompatible segment schemas to use this mode that is external to this PR?

also what about more extreme cases, like sketch aggregators, which have a lot more variety between their building aggregators and combining aggregators?

basically my concern here is around making sure there are no scenarios where we are silently doing the wrong thing, writing incorrect segments, instead of loudly failing when it isn't well supported.

Reply:
- Mix of aggregators with incompatible type. (i.e. compaction with LongSum but some segment has existing metric as String and some has number) is a sad path and will result in the ingestion task failing. The error message will mentioned that this is because isPreserveExistingMetrics is enabled (isPreserveExistingMetrics can be disable to fallback to original behavior). Added this as a test case.
- If non-rolled up has columns with the same name as aggregators in rolled up and isPreserveExistingMetrics is enabled, then those non-rolled up columns would be treated as metrics and combined.
- There is currently no safeguards (other than that the job will fail without creating new unexpected segments). We can look into adding these safeguard later to make the job fail fast. 
- Also tested hll, theta, and quantilesDoublesSketch aggregators via the Integration Tests.